src/components/TextNoteForm.svelte

```diff
@@ -26,16 +30,35 @@
 
 		if (!text) return (posting = false);
 
+		if (replyTo) {
+			const pTags = replyTo.tags.filter((t) => t.startsWith('p»'));
+			tags.push(...pTags);
+			if (!tags.find((t) => t.includes(`p»${replyTo.pubkey}`))) {
+				tags.push(`p»${replyTo.pubkey}`);
+			}
+			let rootId = await getEventRootId(replyTo);
+			if (!rootId) {
+				// is a reply to root
+				rootId = replyTo.id;
+			} else {
+				tags.push(`e»${replyTo.id}»wss://relay.nostr.info»reply`);
+			}
+			tags.push(`e»${rootId}»wss://relay.nostr.info»root`);
```

Comment:
The legacy clients expect e-tags to be ordered in one of these ways:

* no e-tag
* replyTo
* root - replyTo
* root - mentions - replyTo

The code here uses the opposite order for no good reason.

If root==replyTo, only put the root.

Reply:
@louishuddleston this still requires a fix.